skimage/segmentation/active_contour_model.py

```diff
@@ -1,218 +1,249 @@
-    sfree = False
-    if bc.startswith('free'):
-        A[0, :] = 0
-        A[0, :3] = [1, -2, 1]
-        A[1, :] = 0
-        A[1, :4] = [-1, 3, -3, 1]
-        sfree = True
-    efree = False
-    if bc.endswith('free'):
-        A[-1, :] = 0
-        A[-1, -3:] = [1, -2, 1]
-        A[-2, :] = 0
-        A[-2, -4:] = [-1, 3, -3, 1]
-        efree = True
-
-    # Only one inversion is needed for implicit spline energy minimization:
-    inv = np.linalg.inv(A + gamma*np.eye(n))
-
-    # Explicit time stepping for image energy minimization:
-    for i in range(max_iterations):
-        fx = intp(x, y, dx=1, grid=False)
-        fy = intp(x, y, dy=1, grid=False)
-        if sfixed:
-            fx[0] = 0
-            fy[0] = 0
-        if efixed:
-            fx[-1] = 0
-            fy[-1] = 0
-        if sfree:
-            fx[0] *= 2
-            fy[0] *= 2
-        if efree:
-            fx[-1] *= 2
-            fy[-1] *= 2
-        xn = inv @ (gamma*x + fx)
-        yn = inv @ (gamma*y + fy)
-
-        # Movements are capped to max_px_move per iteration:
-        dx = max_px_move*np.tanh(xn-x)
-        dy = max_px_move*np.tanh(yn-y)
-        if sfixed:
-            dx[0] = 0
-            dy[0] = 0
-        if efixed:
-            dx[-1] = 0
-            dy[-1] = 0
-        x += dx
-        y += dy
-
-        # Convergence criteria needs to compare to a number of previous
-        # configurations since oscillations can occur.
-        j = i % (convergence_order+1)
-        if j < convergence_order:
-            xsave[j, :] = x
-            ysave[j, :] = y
-        else:
-            dist = np.min(np.max(np.abs(xsave-x[None, :]) +
-                                 np.abs(ysave-y[None, :]), 1))
-            if dist < convergence:
-                break
-
-    return np.array([x, y]).T
+from warnings import warn
+import numpy as np
+from scipy.interpolate import RectBivariateSpline
+from ..util import img_as_float
+from ..filters import sobel
+
+
+def active_contour(image, snake, alpha=0.01, beta=0.1,
+                   w_line=0, w_edge=1, gamma=0.01,
+                   bc=None, max_px_move=1.0,
+                   max_iterations=2500, convergence=0.1,
+                   *,
+                   boundary_condition='periodic',
+                   coordinates=None):
+    """Active contour model.
+
+    Active contours by fitting snakes to features of images. Supports single
+    and multichannel 2D images. Snakes can be periodic (for segmentation) or
+    have fixed and/or free ends.
+    The output snake has the same length as the input boundary.
+    As the number of points is constant, make sure that the initial snake
+    has enough points to capture the details of the final contour.
+
+    Parameters
+    ----------
+    image : (N, M) or (N, M, 3) ndarray
+        Input image.
+    snake : (N, 2) ndarray
+        Initial snake coordinates. For periodic boundary conditions, endpoints
+        must not be duplicated.
```

Comment:
So these snake coordinates are provided in xy?

Reply:
They are provided in whatever `coordinates=` is.